src/v/utils/hybrid_clock.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "base/likely.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/manual_clock.hh>
+
+#include <chrono>
+
+/// The hybrid_clock is a thin wrapper around the seastar::lowres_clock
+/// that can be swapped from seastar::manual_clock in tests by flipping
+/// a runtime flag (per shard). The timestamps and durations are using
+/// last bit as a flag indicating the type of clock used to produce
+/// them.
+class hybrid_clock {
+    struct duration_with_flag {
+        uint64_t nanoseconds{0};
+        bool manual{false};
+    };
+
```

Comment:
I'm a bit confused about the different modes of this clock. Is the idea that a test would declare the scope to be in manual mode, and then hybrid_clock would act like the manual clock? What's the purpose of the bit hack if we're already keeping track of whether the clock is in manual mode?